File: 409. Longest Palindrome/longestPalindrome.js
/**
 * @param {string} s
 * @return {number}
 */
const longestPalindrome = (s) => {
  const charCnt = {};
  let res = 0;

  for (let i = 0; i < s.length; i++) {
    if (!charCnt[s[i]]) {
      charCnt[s[i]] = 0;
    }
    charCnt[s[i]]++;

    if (charCnt[s[i]] % 2 === 0) {
      res += 2;
    }
  }

  for (let key in charCnt) {
    if (charCnt[key] % 2) {
      res++;
      break;
    }
  }

  return res;
};


const s = "abccccdd";
console.log('longestPalindrome: ', longestPalindrome(s));

File: 409. Longest Palindrome/longestPalindrome.py
class Solution:
  def longestPalindrome(self, s: str) -> int:
    charCnt = {}
    res = 0
    
    for c in s:
      if c not in charCnt:
        charCnt[c] = 0
      charCnt[c] += 1
      
      if charCnt[c] % 2 == 0:
        res += 2

    for cnt in charCnt.values():
      if cnt % 2:
        res += 1
        break
    
    return res


s = "abccccdd"
solution = Solution()
print('longestPalindrome: ', solution.longestPalindrome(s))
